File: Service/MediCare/RMA.Service.MediCare.Contracts/Entities/Medical/TebaInvoiceLine.cs
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;
using System.Collections.Generic;

namespace RMA.Service.MediCare.Contracts.Entities.Medical
{
    public class TebaInvoiceLine : Common.Entities.AuditDetails
    {
        public int TebaInvoiceLineId { get; set; }
        public int TebaInvoiceId { get; set; }
        public System.DateTime ServiceDate { get; set; }
        public decimal? RequestedQuantity { get; set; }
        public decimal AuthorisedQuantity { get; set; }
        public decimal RequestedAmount { get; set; }
        public decimal RequestedVat { get; set; }
        public decimal? RequestedAmountInclusive { get; set; }
        public decimal? AuthorisedAmount { get; set; }
        public decimal? AuthorisedVat { get; set; }
        public decimal? AuthorisedAmountInclusive { get; set; }
        public int TariffId { get; set; }
        public decimal TotalTariffAmount { get; set; }
        public decimal TotalTariffVat { get; set; }
        public decimal? TotalTariffAmountInclusive { get; set; }
        public decimal? TariffAmount { get; set; }
        public decimal CreditAmount { get; set; }
        public VatCodeEnum VatCode { get; set; }
        public decimal? VatPercentage { get; set; }
        public int TreatmentCodeId { get; set; }
        public int MedicalItemId { get; set; }
        public string HcpTariffCode { get; set; }
        public int? TariffBaseUnitCostTypeId { get; set; }
        public string Description { get; set; }
        public int? SummaryInvoiceLineId { get; set; }
        public bool IsPerDiemCharge { get; set; }
        public bool IsDuplicate { get; set; }
        public int DuplicateTebaInvoiceLineId { get; set; }
        public string CalculateOperands { get; set; }
        public List<InvoiceLineUnderAssessReason> InvoiceLineUnderAssessReasons { get; set; }
    }
}

File: Web/RMA-Web-Shell/projects/medicare/src/app/medical-invoice-manager/models/teba-invoice-line.ts
import { BaseClass } from "projects/shared-models-lib/src/lib/common/base-class";
import { InvoiceUnderAssessReason } from "./invoice-under-assess-reason";
import { MedicalInvoiceReport } from "./medical-invoice-report";
import { PreAuthorisation } from "../../preauth-manager/models/preauthorisation";
import { InvoiceLineUnderAssessReason } from "./invoice-line-under-assess-reason";
import { VatCodeEnum } from "../enums/vat-code.enum";

export class TebaInvoiceLine extends BaseClass{
    tebaInvoiceLineId: number;
    tebaInvoiceId: number;
    serviceDate: string;
    requestedQuantity: number | null;
    authorisedQuantity: number;
    requestedAmount: number;
    requestedVat: number;
    requestedAmountInclusive: number | null;
    authorisedAmount: number | null;
    authorisedVat: number | null;
    authorisedAmountInclusive: number | null;
    tariffId: number;
    totalTariffAmount: number;
    totalTariffVat: number;
    totalTariffAmountInclusive: number | null;
    tariffAmount: number | null;
    creditAmount: number;
    vatCode: VatCodeEnum;
    vatPercentage: number | null;
    treatmentCodeId: number;
    medicalItemId: number;
    hcpTariffCode: string;
    tariffBaseUnitCostTypeId: number | null;
    description: string;
    summaryInvoiceLineId: number | null;
    isPerDiemCharge: boolean;
    isDuplicate: boolean;
    duplicateTebaInvoiceLineId: number;
    calculateOperands: string;
    invoiceLineUnderAssessReasons: InvoiceLineUnderAssessReason[];
     //------------not in DB
    validationMark: string;
    //-- include and exclude allocationFields FormArray - FormGroup properties
    exclude?: any;
    include?: any;
    underAssessReason?: string;

}